core/src/main/java/org/geysermc/geyser/util/WebUtils.java

```diff
@@ -96,6 +100,114 @@ public static void downloadFile(String reqURL, String fileLocation) {
         }
     }
 
+    /**
+     * Checks a remote pack URL to see if it is valid
+     * If it is, it will download the pack file and return a path to it
+     *
+     * @param url The URL to check
+     * @param force If true, the pack will be downloaded even if it is cached to a separate location.
+     * @return Path to the downloaded pack file, or null if it was unable to be loaded
+     */
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    public static @Nullable Path downloadRemotePack(String url, boolean force) {
+        GeyserLogger logger = GeyserImpl.getInstance().getLogger();
+        try {
+            HttpURLConnection con = (HttpURLConnection) new URL(url).openConnection();
+
+            con.setConnectTimeout(10000);
+            con.setReadTimeout(10000);
+            con.setRequestProperty("User-Agent", "Geyser-" + GeyserImpl.getInstance().getPlatformType().platformName() + "/" + GeyserImpl.VERSION);
+            con.setInstanceFollowRedirects(true);
+
+            int responseCode = con.getResponseCode();
+            if (responseCode >= 400) {
+                throw new IllegalStateException(String.format("Invalid response code from remote pack at URL: %s (code: %d)", url, responseCode));
+            }
+
+            int size = con.getContentLength();
+            String type = con.getContentType();
+
+            if (size <= 0) {
+                throw new IllegalArgumentException(String.format("Invalid content length received from remote pack at URL: %s (size: %d)", url, size));
+            }
+
+            // This doesn't seem to be a requirement (anymore?). Logging to debug as it might be interesting though.
+            if (type == null || !type.equals("application/zip")) {
+                logger.warning(String.format("Application type received from remote pack at URL %s uses the content type: %s! This may result in packs not loading " +
+                    "for Bedrock players.", url, type));
```

Comment:
In my opinion this should say something along the lines of `Try using a link that is
application/zip type instead!` So that people know what the content type should be instead of being confused on why the content type wont work and not being able to know what the right one is.

Reply:
Hey! Yeah; this is temporary to figure out what application types are (or aren't) supported by Bedrock edition while testing. This will be changed before merging.

So far, it unfortunately seems that only application/zip works - the popular octet-stream type seems to not work. Which is unfortunate, as that's what many services use